process_report/invoices/pi_specific_invoice.py

```diff
@@ -92,20 +92,44 @@ def _get_pi_dataframe(self, data, pi):
             if column_name in pi_projects.columns:
                 column_sums.append(pi_projects[column_name].sum())
                 sum_columns_list.append(column_name)
-        pi_projects.loc[len(pi_projects)] = (
-            None  # Adds a new row to end of dataframe initialized with None
+
+        # Create a new row with proper dtypes
+        new_row = {col: None for col in pi_projects.columns}
+
+        # Add Invoice Month column if it doesn't exist
+        if invoice.INVOICE_DATE_FIELD not in pi_projects.columns:
+            pi_projects[invoice.INVOICE_DATE_FIELD] = None
+            new_row[invoice.INVOICE_DATE_FIELD] = None
+
+        new_row[invoice.INVOICE_DATE_FIELD] = "Total"
+        for col, val in zip(sum_columns_list, column_sums):
+            new_row[col] = val
+
+        # Convert all columns to object type before concatenation to avoid dtype warnings
+        pi_projects = pi_projects.astype("object")
```

Comment:
You should not typecast the dataframe that will be exported into `object` type. Doing so means there will be no type warning or enforcement at all, which we still want. It would also remove the 2 decimal precision that we want on our money columns, so the exported PDFs would have inconsistent precision.

I would suggest instead that you cast the "totals row" to be the same type as `pi_projects` instead:
```suggestion python
new_row = pandas.DataFrame(data={col: [None] for col in pi_projects.columns})
new_row = new_row.astype(pi_projects.dtypes)
pi_projects = pandas.concat([pi_projects, new_row], ignore_index=True)
pi_projects.loc[pi_projects.index[-1], invoice.INVOICE_DATE_FIELD] = "Total"
pi_projects.loc[pi_projects.index[-1], sum_columns_list] = column_sums
...        
```

Reply:
I would also suggest a more informative name than `new_row`. `totals_row` might be better?